domains/infrastructure/front_door.tf

```diff
@@ -5,3 +5,29 @@ resource "azurerm_cdn_frontdoor_profile" "main" {
   sku_name            = "Standard_AzureFrontDoor"
   tags                = var.tags
 }
+
+data "azurerm_monitor_diagnostic_categories" "main" {
+  resource_id = values(azurerm_cdn_frontdoor_profile.main)[0].id
+}
+
+resource "azurerm_monitor_diagnostic_setting" "main" {
+  for_each                   = var.hosted_zone
```

Comment:
We only want to enable it if monitoring is enabled. Could you amend the condition?

Reply:
amended the condition